flagsmith-jira-app/src/frontend/components/IssueFeatureTables.tsx

```diff
@@ -224,29 +224,38 @@ const IssueFeatureTables = ({
 
   return (
     <Fragment>
-      {features
-        .filter((feature) => issueFeatureIds.includes(String(feature.id)))
-        .map((feature, index) => (
-          <Fragment key={feature.id}>
+      {issueFeatureIds.map((featureId) => {
+        const baseFeature = features.find((f) => String(f.id) === featureId);
+        if (!baseFeature) {
+          return null;
+        }
+
+        const envFeaturesForThisFeature = environmentsFeatures
+          .map((envFeatures) => {
+            const matchingFeature = envFeatures.find((f) => String(f.id) === featureId);
+            return matchingFeature;
+          })
+          .filter(Boolean) as Feature[];
```

Comment:
Would you mind extracting the formatting in a specific function (for abstraction) and just map over the result ?

Something that would like:
```
const featureTableData = buildFeatureTableData({
  issueFeatureIds,
  features,
  environmentsFeatures,
});

return (
  <Fragment>
    {featureTableData.map(({ featureId, baseFeature, environmentFeatures }) => (
      <Box key={id} xcss={{ marginTop: "space.300", marginBottom: "space.100" }}>
```